### PR TITLE
Add AttributeListValue

### PR DESCRIPTION
**Pull request by bregman-arie**
This will allows us to set up some of attributes (of the different
models) as Lists of AttributeValue objects.

In addition, add Argument class that will be used by different
attributes of the models, to expose their arguments.

### CODE REVIEW

File: cibyl/models/attribute.py [hunk 1]
```diff
@@ -23,4 +25,39 @@ class AttributeValue():
     name: str
     type: object
```
Comment (cescgina) [inline cibyl/models/attribute.py:26]: type should be attr_type

File: cibyl/models/attribute.py [hunk 2]
```diff
@@ -23,4 +25,39 @@ class AttributeValue():
...
+    arguments: list[Argument]
+
+
+class AttributeListValue(AttributeValue):
+    """Represents a list of AttributeValue objects"""
+
+    def __init__(self, name, arguments=None, attr_type=None, value=None):
+
+        super(AttributeListValue, self).__init__(
+            name, arguments=arguments, attr_type=attr_type)
```
Comment (cescgina) [inline cibyl/models/attribute.py:37]: Missing value parameter

File: cibyl/models/attribute.py [hunk 3]
```diff
@@ -23,4 +25,39 @@ class AttributeValue():
...
+
+        if isinstance(value, list):
+            self.data = value
```
Comment (cescgina) [inline cibyl/models/attribute.py:40]: Are self.data and self.value the same attribute?

**Review by cescgina — CHANGES_REQUESTED** [review]

**Comment by bregman-arie** [comment]
@cescgina thanks for the review! submitted the fixes

File: cibyl/models/attribute.py [hunk 4]
```diff
@@ -15,12 +15,49 @@
...
-    arguments: list
+    arguments: list[Argument]
+
+
+class AttributeListValue(AttributeValue):
+    """Represents a list of AttributeValue objects"""
+
+    def __init__(self, name, arguments=None, attr_type=None, value=None):
+
+        super(AttributeListValue, self).__init__(
```
Comment (cescgina) [inline cibyl/models/attribute.py:36]: Here we still need to pass value to the init method of AttributeValue I think

**Review by cescgina — CHANGES_REQUESTED** [review]
Minor change and should be good to go

File: cibyl/models/attribute.py (hunk 4 above)
Comment (jsanemet) [reply cibyl/models/attribute.py:36]: +1
The value is lost if it is neither a list nor 'None'.

Comment (bregman-arie) [reply cibyl/models/attribute.py:36]: We do, on line 37. Unless I misunderstood you.

File: cibyl/models/attribute.py [hunk 5]
```diff
@@ -15,12 +15,49 @@
...
+        :return:
+        """
+        return self.value[index]
+
+    def __lt__(self, other):
+        """
+        :param other:
+        :return:
+        """
+        return self < other
```
Comment (jsanemet) [inline cibyl/models/attribute.py:63]: I believe that if the operator is to be implemented, then it should be done relative to the attribute's value in some way. Besides, wouldn't this expression lead to a recursive call?

**Review by jsanemet — CHANGES_REQUESTED** [review]

File: cibyl/models/attribute.py (hunk 4 above)
Comment (cescgina) [reply cibyl/models/attribute.py:36]: On the call to the super's init I don't see the value parameter, maybe it was lost when pushing? I only see name, arguments and attr_type passed

Comment (bregman-arie) [reply cibyl/models/attribute.py:36]: oh k got it now. thanks, will fix it

**Review by cescgina — APPROVED** [review]

File: cibyl/models/attribute.py (hunk 5 above)
Comment (bregman-arie) [reply cibyl/models/attribute.py]: Thanks, removed it in the meantime.

**Pull request merged**